### PR TITLE
Adds constructable shuttle turfs.

### PR DESCRIPTION
**Pull request by monster860**
This PR adds new constructable shuttle turfs.
Shuttle walls constructed by placing wall girders, adding rods, then metal.
Shuttle windows are constructed by placing a full rglass window, and adding rods to it before fastening it.
Shuttle floors are made by putting metal rods on floor tiles.

All instances of old shuttle turfs have been replaced with new shuttle turfs with the following exceptions:

- Vox skipjack, syndicate infiltrator
- Arrivals shuttle
- Cargo shuttle
- Crashed shuttles on z-6
- ERT shuttle
- Suspicious Supply Depot
- Emergency shuttle

:cl: monster860
rscadd: Adds a replacement constructable shuttle turfs, and changes the old ones to use this one
rscadd: Makes shuttle windows use icon-smoothing
tweak: Shuttle airlocks and shuttle windows are now constructable
bugfix: Fixes stray /turf/simulated/shuttle/plating on the actual station
bugfix: Fixes some stray /turf/unsimulated/floor instances on the syndicate infiltrator
/:cl:

### CODE REVIEW

**Comment by FlattestGuitar** [comment]
Just to be clear: this doesn't make it possible to alter the preexisting shuttles any more than it's already possible, or make your own moving spacecraft, correct?

**Comment by monster860** [comment]
No, you cannot create a new shuttle. That would require me (or someone else) making a system for allocating transit areas. You can alter the turfs on the pre-existing ones (Breaking/replacing walls), which will make bombing the shuttle more feasible. However, expanding the shuttle is not possible since the docking_port objects are bounded, and it's also bounded by a /area/shuttle.

**Comment by FlattestGuitar** [comment]
Coolio then. :tada:

**Comment by Aurorablade** [comment]
"All instances of old shuttle turfs have been replaced with new shuttle turfs"

.....
No...

File: code/game/turfs/simulated/shuttle.dm [hunk 1]
```diff
@@ -62,4 +62,140 @@
...
+	canSmoothWith = list(
+	/turf/simulated/wall/shuttle,
+	/obj/structure/window/full/shuttle,
+	/obj/machinery/door/airlock,
+	/obj/machinery/door/unpowered,
+	/obj/machinery/door/poddoor,
+	/obj/structure/shuttle)
+	var/fullcorner = 0
+
+/turf/simulated/wall/shuttle/proc/do_old_smooth()
```
Comment (tigercat2000) [inline code/game/turfs/simulated/shuttle.dm]: fuck no, use the new smoothing system

**Comment by Aurorablade** [comment]
okay let me add on to this with:
" which will make bombing the shuttle more feasible"

No..lets not enable asshole shuttle grief more so then some peope go LAWL I HIT NOW right as it is...

File: code/game/turfs/simulated/shuttle.dm (hunk 1 above)
Comment (monster860) [inline code/game/turfs/simulated/shuttle.dm]: That's not feasible right now. The new smoothing system doesn't allow for the use of the corner turfs used in the shuttle turf sprites. In fact, the new smoothing system isn't compatible with shuttle wall sprites at all.

If it makes you feel any better, the WINDOWS use the new smoothing system

**Comment by monster860** [comment]
> No..lets not enable asshole shuttle grief more so then some peope go LAWL I HIT NOW right as it is...

BWOINK.

**Comment by tigercat2000** [comment]
Yeah, fuck no.
:-1: from me for so many reasons, including, but not limited to:
 - Copypasting the old smoothwall code for just the shuttle walls
 - Allowing the shuttles to be bombed more easily

**Comment by monster860** [comment]
@tigercat2000 I'd love to use new smooth code, but the shuttle wall sprites are not compatible with it at all. Even if you get rid of the corner-edge walls it doesn't work. And it simply just doesn't work WITH corner-edge walls, which I assume you want to keep

**Comment by Aurorablade** [comment]
Scenario: Somone goes to effort to reconstruct escape shuttle...escape shuttle is called...people rush into shuttle in panic without realizing..actual escape shuttle arrives and gibs the lot of them.

**Comment by monster860** [comment]
@Aurorablade The kind of thing a traitor-engi with a hijack objective wouild do. That's actually a pretty great idea.

**Comment by Mijhead** [comment]
It's also pretty funny.

**Comment by Fox-McCloud** [comment]
:-1: 

No. This needs to use the existing system AND the shuttle walls absolutely should not be destructible.

**Comment by ScratKnapp** [comment]
Wouldn't it be possible to make a specific creatable shuttle-tile that can be deconstructed and leave the tiles used for actual shuttles as a seperate, still invincible type?

**Comment by monster860** [comment]
@Fox-McCloud Why should shuttle walls be indestructable? That's not realistic at all.

**Comment by Spacemanspark** [comment]
Realism < fun.
That said, the general concept on this PR is alright, just implement what Fox/ Tiger said.

**Comment by monster860** [comment]
If some asshole kabooms the shuttle, thats griefing. What you can now do is get to the asshole who telescis himself into the wall.

**Comment by monster860** [comment]
Is this better?

**Comment by Crazylemon64** [comment]
Since administration lacks tools to quickly repair the shuttle, we'll still have issues with people blowing it up - perhaps a new buildmode tool might be handy here... I think I'll try doing that.

**Comment by monster860** [comment]
@Crazylemon64 

I disagree - Using adv buildmode, and rembering the paths to shuttle turfs is not too hard off the top of my head. Hell, I can remember icon_states off the top of my head.

**Comment by Crazylemon64** [comment]
@monster860 The shuttle docks for only 3 minutes - there's not much time to finish the work in, and it must be done manually, object-per-object, especially if some time's already elapsed until the bomber does their damage.

**Comment by monster860** [comment]
You have a point there.

On a side note, this would make building new shuttles easier by adminbus. Modifying icon_states is tedious, and so is placing underlays manually.

*By the way, I am planning on a buildable shuttles PR, but that is for later. And no, you won't be able to summon a shuttle onto the middle of the medbay/bar/whatever area. There are however many things I need to do between this PR and that future PR.*

**Comment by Necaladun** [comment]
I am against allowing the escape shuttle to be fucked with. That'll just be a problem 100 times for every 1 time it's actually useful.

**Comment by monster860** [comment]
@Fox-McCloud I believe I've addressed all your concerns here.

File: code/game/objects/structures/girders.dm [hunk 2]
```diff
@@ -29,7 +29,7 @@
 		qdel(src)
 
 /obj/structure/girder/attackby(obj/item/W as obj, mob/user as mob, params)
-	if(istype(W, /obj/item/weapon/wrench) && state == 0)
+	if(istype(W, /obj/item/weapon/wrench) && state == 0 && icon_state != "shuttlegirder")
```
Comment (TheDZD) [inline code/game/objects/structures/girders.dm]: Do not use `icon_state`s for this, it's hacky.

**Comment by TheDZD** [comment]
I just honestly don't like the idea of players being able to fuck with shuttles.

**Comment by monster860** [comment]
The point of this is to eventually get to constructable moving shuttles. Until then, this will probably be used by *Fungineers* to make the station look spiffy.

**Comment by Aurorablade** [comment]
> The point of this is to eventually get to constructable moving shuttles. 

...That will make things worse...

**Comment by TheDZD** [comment]
@monster860 That falls under the scope of fucking with shuttles.

**Comment by monster860** [comment]
Soooooo @TheDZD @Fox-McCloud  ??

**Comment by Rumiluntti** [comment]
This seems like a horrible idea in its entirety

**Comment by monster860** [comment]
Why?

The only shuttle that really needs to be invincible is the escape shuttle, and it has a reason for being that way. (Also arrivals because that's where players start and we don't want that getting destroyed)

I have kept the escape and arrivals shuttles intact. All the other shuttles, however, we should be able to play with. There's no reason for the white ship to be an invincible fort with unbreakable walls, and same with just about every other shuttle in existence.

Allow me to quote the wiki
>Weak Points
>Judge how high security the room will be, if it is high security, reinforced walls and electrified grill windows may be in order. Areas that do not need a lot of security can use basic walls, and windows to your liking (though normal glass windows break very very easy). Each room should have one place that's weaker than the rest (like a back door, side entrance, or a window), just because the main entrance might be out of commission (and realistically, for traitors to break into).

There's no reason for places like the escape pods to have invincible walls. Any old fucker can just waltz their ass on to one and security won't give a shit. While the research and mining shuttles do have access restrictions, they are still low-security areas. However, they have invincible walls.

**Comment by TheDZD** [comment]
As already discussed above, closing this PR. We don't believe players should have access to shuttle construction, and so allowing them to build shuttle turfs doesn't really make sense, especially since this would touch systems outside of just shuttle turfs.

**Pull request closed without merging**